integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -71,12 +74,46 @@ def __init__(self, user, enterprise_configuration):
 
     def export(self, **kwargs):
         """
-        Return the exported and transformed content metadata as a dictionary.
+        Export transformed content metadata if there has been an update to the consumer's catalogs
+        """
+        enterprise_customer_catalogs = self.enterprise_configuration.customer_catalogs_to_transmit or \
+            self.enterprise_customer.enterprise_customer_catalogs.all()
+
+        catalogs_to_transmit = []
+        catalogs_last_modified = {}
+        for enterprise_customer_catalog in enterprise_customer_catalogs:
+            enterprise_catalog = self.enterprise_catalog_api.get_enterprise_catalog(enterprise_customer_catalog.uuid)
+            if not enterprise_catalog.get('content_last_modified'):
+                if enterprise_catalog.get('catalog_modified'):
+                    catalog_last_modified = enterprise_catalog.get('catalog_modified')
+                else:
+                    catalog_last_modified = None
+            elif not enterprise_catalog.get('catalog_modified'):
+                catalog_last_modified = enterprise_catalog.get('content_last_modified')
+            else:
+                catalog_last_modified = max(
+                    enterprise_catalog.get('content_last_modified'),
+                    enterprise_catalog.get('catalog_modified')
+                )
```

Comment:
This if block is breaking my brain.
So after my 5th or so read, if I'm reading this right, why not just do like:
```
if not (enterprise_catalog.get('content_last_modified') AND enterprise_catalog.get('catalog_modified'):
    catalog_last_modified = None
else:
   catalog_last_modified = max(
          enterprise_catalog.get('content_last_modified', '0'),
          enterprise_catalog.get('catalog_modified', '0')
    )
```
(Ignore my hideous space/indentation count, I'm writing this on github)

Reply:
(From the test input it seems like we get a string at this stage which is why I use a default stringified 0 instead of say an int or None since that breaks the max operation)

Reply:
Lol yeah this was as much a pain to write as it is probably to read. My only concern is comparing `0` to datetimes, but this could potentially be a great improvement to readability.

Reply:
are they true datetimes though? From the test data they appear to be strings that _can_ be read in as datetime by a method that parses them that way. We don't seem to convert them to actual datetimes anywhere up to this point as far as I could tell. Thus string comparisons :p

Reply:
@alex-sheehan-edx I'd suggest forcing `catalog_last_modified` and `last_successful_transmission` to always be datetime objects to make this whole thing simpler.  

* Have `_get_most_recent_catalog_update_time()` return `datetime.min` by default instead of `None`.
* Deserialize `content_last_modified` and `catalog_modified` to datetimes inside of `enterprise_catalog_api get_enterprise_catalog()` - you can probably use `dateparse.parse_datetime()`.  Set them to `datetime.min` if they're not present in the response.

After doing that, you can just do
```python
catalog_last_modified = max(
      enterprise_catalog['content_last_modified'],
      enterprise_catalog['catalog_modified'],
)
last_successful_transmission = self._get_most_recent_catalog_update_time()
if last_successful_transmission < catalog_last_modified:
    # do your thing
```

Reply:
Unresolving this because I agree with Talia that the crazy if statements hurt my brain.